# SPATAC-seq: **s**plit-**p**ool ligation-based scATAC-seq method
## Overview of the molecular structure of SPATAC-seq

![Image text](https://github.com/Lan-lab/SPATAC-seq/blob/main/SPATAC_workflow.png)

## Dependiencies

* Python 3.0

* [FASTX-Toolkit](http://hannonlab.cshl.edu/fastx_toolkit/commandline.html)
* [Cell Ranger ATAC v1.2](https://support.10xgenomics.com/single-cell-atac/software/downloads/1.2/)
* [Sinto](https://timoast.github.io/sinto/index.html)
* [Signac](https://github.com/stuart-lab/signac)
* [ArchR](https://github.com/GreenleafLab/ArchR)

## Citation

1. Cusanovich DA, Daza R, Adey A, et al. [Multiplex single cell profiling of chromatin accessibility by combinatorial cellular indexing](https://www.science.org/doi/10.1126/science.aab1601). *Science*. 2015;348(6237):910-914. doi:10.1126/science.aab1601.
2. Amini S, Pushkarev D, Christiansen L, et al. [Haplotype-resolved whole-genome sequencing by contiguity-preserving transposition and combinatorial indexing](https://www.nature.com/articles/ng.3119). *Nat Genet*. 2014;46(12):1343-1349. doi:10.1038/ng.3119.
3. Rosenberg AB, Roco CM, Muscat RA, et al. [Single-cell profiling of the developing mouse brain and spinal cord with split-pool barcoding](https://www.science.org/doi/10.1126/science.aam8999). *Science*. 2018;360(6385):176-182. doi:10.1126/science.aam8999.
4. Ma S, Zhang B, LaFave LM, et al. [Chromatin Potential Identified by Shared Single-Cell Profiling of RNA and Chromatin](https://www.sciencedirect.com/science/article/pii/S0092867420312538). *Cell*. 2020;183(4):1103-1116.e20. doi:10.1016/j.cell.2020.09.056.